### ISSUE TITLE
VoiceOver doesn't always navigate Radio Button Group successfully 

### ISSUE THREAD

**Issue by phinch** [grommet/grommet#7002]
Hi! When using VoiceOver with a radio button group, and using VoiceOver commands to navigate through the labels, we're running into some strange behavior where the focus is sometimes forced back to the first radio button. I work on an enterprise software team, and our accessibility auditors have told us that this is violating our accessibility requirements :( 

### Expected Behavior

When viewing a radio button group, the user should be able to scan through the radio buttons and labels by using the VoiceOver arrow key commands.

### Actual Behavior

When moving through the radio button group using the VoiceOver VO+→ and VO+←, the focus occasionally jumps back to the first radio button, which takes the user off of their focused element and back to the beginning of the radio group.

I'm wondering if the root cause is [this imperative focus behavior in the radio group button](https://github.com/grommet/grommet/blob/master/src/js/components/RadioButtonGroup/RadioButtonGroup.js#L71-L77)...it seems like this effect would fire often, since i think onFocus and onBlur are firing often as we move through the component in VO, and since `valueIndex` is 0 when the first option is selected, we would satisfy the condition where `(focus && !valueIndex)` and force focus back to the first radio button? But i'm not sure i'm fully understanding all the behavior there.
![image](https://github.com/grommet/grommet/assets/9158038/e886a133-2463-43ed-9268-399ea528d4a4)

### URL, screen shot, or Codepen exhibiting the issue

I was able to reproduce this issue on the [RadioGroup example on the Grommet website](https://v2.grommet.io/radiobuttongroup?) The steps might be hard to communicate because it requires VoiceOver (and tbh i'm not a VO expert), so I'll also attach a video that hopefully shows the behavior.

### Steps to Reproduce

1. Focus on the radio group
1. Use VO+→ and and VO+← to move through the buttons and labels of the component
1. Occasionally, notice that the focus shifts back to the first radio button

#### Reproduction example

At 0:08 and again at 0:19, you can see the focus jumps back to the first radio button even though i'm scanning the second element of the radio group

(As a bonus, our accessibility auditors pointed out that ideally, we should add `focusable=false aria-hidden=true` to the "checked" icon that shows up when a button is selected, so that the screenreader doesn't read out the additional "image" text when scanning through the options - but that's less critical than the forced focus issue)

https://github.com/grommet/grommet/assets/9158038/401150af-d064-4925-aeef-0522732a2996

### Your Environment

- Grommet version: latest
- Browser Name and version: Chrome 118.0.5993.96
- Operating System and version (desktop or mobile): Desktop

Our team was also able to reproduce this on mobile using iOS VoiceOver, but i did more extensive testing on desktop

**Comment by jcfilben** [comment]
Hi @phinch thanks for flagging this issue. I think you have the right idea that this is being caused by the code here: https://github.com/grommet/grommet/blob/master/src/js/components/RadioButtonGroup/RadioButtonGroup.js#L71-L77. I think when valueIndex is 0 this statement is evaluated as true `if (focus && !valueIndex)` because the `!` operator on something with a 0 value will evaluate as true. I think if we adjust this statement to check if valueIndex is actually undefined then we won't run into this issue.

**Comment by phinch** [comment]
@jcfilben thanks for looking into it! i think that makes sense 👍 i was wondering if there might still be a focus issue in the case  where no option is selected yet? it seems like in that case `valueIndex` would legitimately be undefined, but i imagine we'd still want a VoiceOver user to be able to scan through the labels in the radio before selecting an option

**Comment by jcfilben** [comment]
@phinch great note, I think we would run into a focus issue if no option is selected. Looking at this useEffect more it seems like we may not actually need it. Based on the comment in the code it seems like the useEbffect function's intent is to ensure the focus is placed on the first RadioButton when the RadioButtonGroup receives focus if there is no option selected. However it seems like this is already happening by default. Perhaps we can try removing the useEffect

**Comment by phinch** [comment]
@jcfilben ah ok! that sounds like it could work 😄 i'm (maybe obviously) not familiar at all with the various effects and timeouts in this component, so i'm not sure what the risk of removing that effect would be - but it sounds like a good idea to me

**Comment by phinch** [comment]
@jcfilben sorry to pester - would you have a general idea of when the grommet team might be able to prioritize this issue (assuming we don't get any outside contributors like me helping out here haha)? the team i'm working with is asking for a general estimate of when we might get a fix (which would help us determine whether we should try to implement some workaround on our side in the meantime)

**Comment by jcfilben** [comment]
Hi @phich this was a pretty quick so I went ahead and just put up a PR for it

**Comment by phinch** [comment]
@jcfilben yay, thanks so much! i (and everyone on our team) really appreciate it 😄
